packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -32,22 +32,36 @@ export class ExtensionService {
             return;
         }
 
+        const disabledPreloadRegions = [
+            ExtensionRegion.SummaryAfter,
+            ExtensionRegion.SummaryLastItemAfter,
+        ];
+
         extensions?.forEach((extension) => {
+            if (disabledPreloadRegions.includes(extension.region)) {
+                return;
+            }
+
             const url = new URL(extension.url);
 
             url.searchParams.set('extensionId', extension.id);
             url.searchParams.set('cartId', cartId);
             url.searchParams.set('parentOrigin', parentOrigin);
 
-            const link = document.createElement('link');
+            const iframe = document.createElement('iframe');
+
+            iframe.src = url.toString();
+            iframe.style.display = 'none';
 
-            link.rel = 'preload';
-            link.as = 'document';
-            link.href = url.toString();
+            const removeIframe = () => {
+                if (iframe.parentNode) {
+                    iframe.parentNode.removeChild(iframe);
```

Comment:
Instead of loading the iframe in a temporary location, removing it, and then adding it back in the actual location, is it possible to just hide and show iframe that should be preloaded. This way we don't have the limitation of needing TTL caching, which may not work for extensions that have dynamic content.

Reply:
> just hide and show iframe that should be preloaded

Thanks for your feedback @davidchin.

It appears that when `checkout-js` renders a region's container or its parent component (e.g., the selected shipping method region), that's when we need to present the extension. 

Therefore, using the hide and show method might not have the same effect as loading an extension at page load.

Additionally, we previously tested that moving an iframe to a different DOM element results in reloading, so the hide and show approach won't work in this case either.

Reply:
I thought the intention is to load the extension in the background and _show_ it immediately using CSS. As long as we don't attach the iframe to a different parent, it shouldn't reload. I suppose the trick is to make the extension region persistent so they don't remount when checkout steps are collapsed and expanded.

The concern I have with the current approach is that it may have a performance benefit for those extensions that can utilise TTL cache by warming up the cache, but it'd have a performance impact on those extensions that can't as they need to be loaded twice.

Reply:
> load the extension in the background and show it immediately using CSS. 


> the trick is to make the extension region persistent so they don't remount when checkout steps are collapsed and expanded.

I explored this during an early spike trial. After positioning the region with CSS absolute positioning, the major challenge was creating a placeholder that occupies the same space as the extension in the actual area (e.g., a shipping method) and keeping the size in sync whenever either side changes its size. This approach felt hacky.

I agree the performance impact concern is valid. So far, we have only applied pre-loading to 3 out of 5 regions, and they are all in the shipping step. Developers will be informed that setting a TTL cache header would allow this extension to appear like a native component. In the future, we might introduce an API flag to determine if an extension wants pre-loading.

cc @bc-charlesho

Reply:
> After positioning the region with CSS absolute positioning, the major challenge was creating a placeholder that occupies the same space as the extension in the actual area

I understand if the parent of an extension region needs to be unmounted/re-mounted, we won't be able to simply hide and show the extension. We will need to use a workaround like what you tried before. Another idea that I think we can try is to lift up the extension regions so they are closer to the root node (i.e.: `checkout`). That way we can avoid unmounting/re-mounting the regions.